Rules/AvoidUsingDeprecatedManifestFields.cs

```diff
@@ -0,0 +1,126 @@
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+using System;
+using System.Collections.Generic;
+using System.Management.Automation.Language;
+using System.Management.Automation;
+using Microsoft.Windows.Powershell.ScriptAnalyzer.Generic;
+using System.ComponentModel.Composition;
+using System.Globalization;
+
+namespace Microsoft.Windows.Powershell.ScriptAnalyzer.BuiltinRules
+{
+    /// <summary>
+    /// AvoidUsingDeprecatedManifestFields: Run Test Module Manifest to check that no deprecated fields are being used.
+    /// </summary>
+    [Export(typeof(IScriptRule))]
+    public class AvoidUsingDeprecatedManifestFields : IScriptRule
+    {
+        /// <summary>
+        /// AnalyzeScript: Run Test Module Manifest to check that no deprecated fields are being used.
+        /// </summary>
+        /// <param name="ast">The script's ast</param>
+        /// <param name="fileName">The script's file name</param>
+        /// <returns>A List of diagnostic results of this rule</returns>
+        public IEnumerable<DiagnosticRecord> AnalyzeScript(Ast ast, string fileName)
+        {
+            if (ast == null)
+            {
+                throw new ArgumentNullException(Strings.NullAstErrorMessage);
+            }
+
+            if (String.Equals(System.IO.Path.GetExtension(fileName), ".psd1", StringComparison.OrdinalIgnoreCase))
+            {
+                var ps = System.Management.Automation.PowerShell.Create(RunspaceMode.CurrentRunspace);
+                IEnumerable<PSObject> result = null;
+                try
+                {
+                    ps.AddCommand("Test-ModuleManifest");
+                    ps.AddParameter("Path", fileName);
+
+                    // Suppress warnings emitted during the execution of Test-ModuleManifest
+                    // ModuleManifest rule must catch any violations (warnings/errors) and generate DiagnosticRecord(s)
+                    ps.AddParameter("WarningAction", ActionPreference.SilentlyContinue);
+                    ps.AddParameter("WarningVariable", "Message");
+                    ps.AddScript("$Message");
+                    result = ps.Invoke();
+
+                }
+                catch
+                {}
+
+                if (result != null)
+                {
+                    foreach (var warning in result)
+                    {
+                       yield return new DiagnosticRecord(String.Format(CultureInfo.CurrentCulture, warning.BaseObject.ToString()), ast.Extent, GetName(), DiagnosticSeverity.Warning, fileName);
```

Comment:
Is there going to be a null pointer exception if the baseobject is null? If I remember correctly, we got this bug somewhere before.